### PR TITLE
Moves more logic to DOMEditor.

### PR DESCRIPTION
**Pull request by diegoreymendez**
<h3>Description:</h3>

Moves additional logic from `ElementNode` into `DOMEditor`.

Solves part of #341 .

<h3>How to test:</h3>

**Test 1:**

Run the unit tests.

**Test 2:**

Run the editor and try editing.  Make sure to compare any flaws found against develop.

### CODE REVIEW

File: Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift [hunk 1]
```diff
@@ -16,6 +20,279 @@ extension Libxml2 {
...
+                return child
+            })
+
+            element.wrap(children: children, inElement: elementDescriptor)
+        }
+
+        /// Wraps the specified range inside a node with the specified properties.
+        ///
+        /// - Parameters:
+        ///     - targetRange: the range that must be wrapped.
```
Comment (SergioEstevao) [inline Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift]: missing the doc for the element argument.

File: Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift [hunk 2]
```diff
@@ -16,6 +20,279 @@ extension Libxml2 {
...
+        }
+
+        /// Wraps child nodes intersecting the specified range inside new elements with the
+        /// specified properties.
+        ///
+        /// - Parameters:
+        ///     - targetRange: the range that must be wrapped.
+        ///     - elementDescriptor: the descriptor for the element to wrap the range in.
+        ///
+        func wrapChildren(of element: ElementNode, intersectingRange targetRange: NSRange, inElement elementDescriptor: ElementNodeDescriptor) {
```
Comment (SergioEstevao) [inline Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift]: Missing the documentation for the element argument.

File: Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift [hunk 3]
```diff
@@ -16,6 +20,279 @@ extension Libxml2 {
...
+        ///
+        func unwrap(range: NSRange, fromElementsNamed elementNames: [String]) {
+            unwrap(rootNode, range: range, fromElementsNamed: elementNames)
+        }
+
+        /// Unwraps the specified range from nodes with the specified name.  If there are multiple
+        /// nodes with the specified name, the range will be unwrapped from all of them.
+        ///
+        /// - Parameters:
+        ///     - range: the range that must be unwrapped.
```
Comment (SergioEstevao) [inline Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift]: Missing the documentation for the element argument.

File: Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift [hunk 4]
```diff
@@ -16,6 +20,279 @@ extension Libxml2 {
...
+                }
+                
+                element.unwrapChildren()
+            }
+        }
+
+        /// Unwraps all child nodes from elements with the specified names.
+        ///
+        /// - Parameters:
+        ///     - range: the range we want to unwrap.
```
Comment (SergioEstevao) [inline Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift]: Missing the documentation for the element argument.

File: Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift (hunk 3 above)
Comment (diegoreymendez) [inline Aztec/Classes/Libxml2/DOM/Logic/DOMEditor.swift]: In this particular case there's no element param, but I'm adding one in the other ones. Good findings!

**Comment by diegoreymendez** [comment]
Ty @SergioEstevao !

**Pull request merged**